tests/vmipreset_test.go

```diff
@@ -297,21 +297,59 @@ var _ = Describe("[rfe_id:609][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 		It("[test_id:1605]should denied to start the VMI", func() {
 			err := virtClient.RestClient().Post().Resource("virtualmachineinstancepresets").Namespace(tests.NamespaceTestDefault).Body(conflictPreset).Do().Error()
 			Expect(err).ToNot(HaveOccurred())
-			waitForPreset(virtClient, conflictPrefix)
+			waitForPreset(virtClient, conflictPrefix, true)
 
 			err = virtClient.RestClient().Post().Resource("virtualmachineinstancepresets").Namespace(tests.NamespaceTestDefault).Body(memoryPreset).Do().Error()
 			Expect(err).ToNot(HaveOccurred())
-			waitForPreset(virtClient, memoryPrefix)
+			waitForPreset(virtClient, memoryPrefix, true)
 
 			vmi.Labels = map[string]string{flavorKey: memoryFlavor, conflictKey: conflictFlavor}
 			By("creating the VirtualMachineInstance")
 			_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
 			Expect(err).To(HaveOccurred())
 		})
 	})
+
+	Context("Preset Lifecycle", func() {
+		var preset *v1.VirtualMachineInstancePreset
+		presetNamePrefix := "vmi-preset-small-"
+		selectorKey := "kubevirt.io/vmPreset"
+		selectorLabel := "vmi-preset-small"
+
+		BeforeEach(func() {
+			selector := k8smetav1.LabelSelector{MatchLabels: map[string]string{selectorKey: selectorLabel}}
+			memory, _ := resource.ParseQuantity("64M")
+			preset = &v1.VirtualMachineInstancePreset{
+				ObjectMeta: k8smetav1.ObjectMeta{GenerateName: presetNamePrefix},
+				Spec: v1.VirtualMachineInstancePresetSpec{
+					Selector: selector,
+					Domain: &v1.DomainSpec{
+						Resources: v1.ResourceRequirements{Requests: k8sv1.ResourceList{
+							"memory": memory}},
+					},
+				},
+			}
+		})
+
+                It("[test_id:617][rfe_id:609] should create and delete preset", func() {
+			By("Creating preset")
+			err := virtClient.RestClient().Post().Resource("virtualmachineinstancepresets").Namespace(tests.NamespaceTestDefault).Body(preset).Do().Error()
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking that preset was created")
+			newPreset := waitForPreset(virtClient, presetNamePrefix, true)
+
+			By("Deleting preset")
+			err = virtClient.RestClient().Delete().Resource("virtualmachineinstancepresets").Namespace(tests.NamespaceTestDefault).Name(newPreset.GetName()).Do().Error()
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking preset was deleted")
+			waitForPreset(virtClient, presetNamePrefix, false)
```

Comment:
Why not have an Eventually loop that tries to Get the preset until DoesNotExist error is encountered?

Reply:
waitForPreset *is* an Eventually loop that tries to get the preset until DoesNotExist error is encountered.